File: GardenTracker.Core/Entities/MarketPrice.cs
using GardenTracker.Core.Enums;

namespace GardenTracker.Core.Entities;

public class MarketPrice
{
    public int Id { get; set; }
    public int SeasonId { get; set; }
    public int PlantTypeId { get; set; }
    public int? PlantVarietyId { get; set; }
    public decimal PricePerUnit { get; set; }
    public HarvestUnit Unit { get; set; }
    public DateOnly RecordedDate { get; set; }

    public Season Season { get; set; } = null!;
    public PlantType PlantType { get; set; } = null!;
    public PlantVariety? PlantVariety { get; set; }

    // Populated by Dapper JOINs — not persisted
    public string? PlantTypeName { get; set; }
    public string? PlantVarietyName { get; set; }
}

File: GardenTracker.Web/src/types/marketPrice.ts
import type { HarvestUnit } from './harvest'

export interface MarketPrice {
  id: number
  seasonId: number
  plantTypeId: number
  plantTypeName: string
  plantVarietyId?: number | null
  plantVarietyName?: string | null
  pricePerUnit: number
  unit: HarvestUnit
  recordedDate: string
}

export interface CreateMarketPriceRequest {
  plantTypeId: number
  plantVarietyId?: number | null
  pricePerUnit: number
  unit: HarvestUnit
  recordedDate: string
}
